runtime/vm/resolvesupport.cpp

```diff
@@ -1839,6 +1839,102 @@ resolveMethodHandleRef(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA cpInde
 
 }
 
+j9object_t
+resolveConstantDynamic(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA cpIndex, UDATA resolveFlags)
+{
+	J9RAMConstantDynamicRef *ramCPEntry = (J9RAMConstantDynamicRef*)ramCP + cpIndex;
+	j9object_t value = ramCPEntry->value;
+
+	J9Class *ramClass = ramCP->ramClass;
+	J9ROMClass *romClass = ramClass->romClass;
+	J9ROMConstantDynamicRef *romConstantRef = (J9ROMConstantDynamicRef*)(J9_ROM_CP_FROM_CP(ramCP) + cpIndex);
+	J9SRP *callSiteData = (J9SRP *) J9ROMCLASS_CALLSITEDATA(romClass);
```

Comment:
you can declare callSiteData, bsmIndices, bsmData, bsmIndex and nameAndSig later since they are only being used in the inner scope

Reply:
same with romClass and romConstantRef